pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go

```diff
@@ -241,6 +241,162 @@ func getLoadbalancerByName(client *gophercloud.ServiceClient, name string) (*loa
 	return &loadbalancerList[0], nil
 }
 
+func getListenersByLoadBalancerID(client *gophercloud.ServiceClient, id string) ([]listeners.Listener, error) {
+	var existingListeners []listeners.Listener
+	err := listeners.List(client, listeners.ListOpts{LoadbalancerID: id}).EachPage(func(page pagination.Page) (bool, error) {
+		listenerList, err := listeners.ExtractListeners(page)
+		if err != nil {
+			return false, err
+		}
+		existingListeners = append(existingListeners, listenerList...)
+
+		return true, nil
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return existingListeners, nil
+}
+
+// get listener for a port or nil if does not exist
+func getListenerForPort(existingListeners []listeners.Listener, port api.ServicePort) (*listeners.Listener) {
+	for _, l := range existingListeners {
+		if l.Protocol == string(port.Protocol) && l.ProtocolPort == int(port.Port) {
+			return &l
+		}
+	}
+
+	return nil
+}
+
+// Get pool for a listener. A listener always has exactly one pool.
+func getPoolByListenerID(client *gophercloud.ServiceClient, loadbalancerID string, listenerID string) (*v2_pools.Pool, error) {
+	listenerPools := make([]v2_pools.Pool, 0, 1)
+	err := v2_pools.List(client, v2_pools.ListOpts{LoadbalancerID: loadbalancerID}).EachPage(func(page pagination.Page) (bool, error) {
```

Comment:
We've just discovered an preexisting issue with the LBaaS v2 API+K8. It is not possible (at least in Liberty) to filter the pool list by loadbalancer, or listener, ID. Neutron will silently ignore the parameter and return all pools from all listeners. 

Using this PR, Liberty OpenStack:

* Create 2 LB services.
* Wait for LB's to up+ready
* Inspect the LBaaS Pool list - Verify there are 2 pools, 1 for each LB.
* Delete 1 of the LB services.
* Inspect the LBaaS Pool list - Verify there 0 pools - pools for both loadbalancers have been deleted.

(Don't you just love Neutron's insistence on ignoring unexpected parameters)